smplrout.cc

```diff
@@ -71,136 +72,75 @@
+                               wpt3->latitude, wpt3->longitude));
     break;
   case metric_t::length:
-    xte_rec->distance = radtomiles(
-                          gcdist(wpt1->latitude, wpt1->longitude,
-                                 wpt3->latitude, wpt3->longitude) +
-                          gcdist(wpt3->latitude, wpt3->longitude,
-                                 wpt2->latitude, wpt2->longitude) -
-                          gcdist(wpt1->latitude, wpt1->longitude,
-                                 wpt2->latitude, wpt2->longitude));
+    track_error = radtomiles(
+                    gcdist(wpt1->latitude, wpt1->longitude,
+                           wpt3->latitude, wpt3->longitude) +
+                    gcdist(wpt3->latitude, wpt3->longitude,
+                           wpt2->latitude, wpt2->longitude) -
+                    gcdist(wpt1->latitude, wpt1->longitude,
+                           wpt2->latitude, wpt2->longitude));
     break;
   case metric_t::relative:
-    if (wpt3->hdop == 0) {
-      fatal(MYNAME ": relative needs hdop information.\n");
-    }
+  default: // eliminate false positive warning with g++ 11.3.0: ‘error’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     // if timestamps exist, distance to interpolated point
-    if (wpt1->GetCreationTime().isValid() && 
+    if (wpt1->GetCreationTime().isValid() &&
         wpt2->GetCreationTime().isValid() &&
         wpt3->GetCreationTime().isValid() &&
         (wpt1->GetCreationTime() != wpt2->GetCreationTime())) {
       double frac = static_cast<double>(wpt1->GetCreationTime().msecsTo(wpt3->GetCreationTime())) /
                     static_cast<double>(wpt1->GetCreationTime().msecsTo(wpt2->GetCreationTime()));
+      double reslat, reslon;
       linepart(wpt1->latitude, wpt1->longitude,
                wpt2->latitude, wpt2->longitude,
                frac, &reslat, &reslon);
-      xte_rec->distance = radtometers(gcdist(
-                                        wpt3->latitude, wpt3->longitude,
-                                        reslat, reslon));
+      track_error = radtometers(gcdist(
+                                  wpt3->latitude, wpt3->longitude,
+                                  reslat, reslon));
     } else { // else distance to connecting line
-      xte_rec->distance = radtometers(linedist(
-                                        wpt1->latitude, wpt1->longitude,
-                                        wpt2->latitude, wpt2->longitude,
-                                        wpt3->latitude, wpt3->longitude));
+      track_error = radtometers(linedist(
+                                  wpt1->latitude, wpt1->longitude,
+                                  wpt2->latitude, wpt2->longitude,
+                                  wpt3->latitude, wpt3->longitude));
     }
     // error relative to horizontal precision
-    xte_rec->distance /= (6 * wpt3->hdop);
+    track_error /= (6 * wpt3->hdop);
```

Comment:
That URL is no longer live, so it doesn't really explain anything. 

As arbitrary as GPS DOP is, I wonder if "6" is just an equally arbitrary magnifier, but since it's constant to all of them, does it really scale anything at all?

Reply:
Its worse than that.  SimplifyRouteFilter::init converts the error option value to miles.  For the relative option, SimplifyRouteFilter::compute_xte computes the incremental change in meters.  I noticed this but didn't change it.

Reply:
> Its worse than that.

It always is... :-)

Thanx for the whole she-bang.